libafl/src/corpus/ondisk.rs

```diff
@@ -191,15 +208,23 @@ where
     where
         P: AsRef<Path>,
     {
-        Self::_new(dir_path.as_ref(), None)
+        Self::with_meta_format_and_path(dir_path.as_ref(), None, None)
     }
 
-    /// Private fn to crate a new corpus at the given (non-generic) path with the given optional `meta_format`
-    fn _new(dir_path: &Path, meta_format: Option<OnDiskMetadataFormat>) -> Result<Self, Error> {
+    /// Creates a new corpus at the given (non-generic) path with the given optional `meta_format`
+    /// and `prefix`.
+    ///
+    /// Will error, if [`std::fs::create_dir_all()`] failed for `dir_path`.
+    pub fn with_meta_format_and_path(
```

Comment:
An API with `Option` shouldn't be public, not very rusty imho

Reply:
In some cases it makes more sense, e.g. when we're parsing these command line parameters. I can undo this change if you wish, but it does simplify my life quite a bit since I want to expose a libfuzzer-style parameter for meta format.

Reply:
So you want to change the metadata according to a commandline flag?
You could always just use an if statement? But sure, if you think it's needed (and clean) as API, we can do it

Reply:
I think it's simpler; the alternative would be (in my options parser):
```rs
match (options.meta_format(), options.path()) {
  (Some(a), Some(b)) => OnDiskCorpus::with_meta_format_and_path(..., a, b),
  (Some(a), None) => OnDiskCorpus::with_meta_format(..., a),
  ...
}
```

Introduces a lot of frustrating branching since we ultimately wrap it in an option anyways.